### PR TITLE
SPSH-1275

### PR DESCRIPTION
**Pull request by DeFi3298**
Ticket: https://ticketsystem.dbildungscloud.de/browse/SPSH-1275

### CODE REVIEW

**Review by tikle-cap — APPROVED** [review]

**Review by he-meyer — APPROVED** [review]

File: src/modules/person/persistence/person.repository.integration-spec.ts [hunk 1]
```diff
@@ -113,6 +115,10 @@ describe('PersonRepository Integration', () => {
                     provide: KeycloakUserService,
                     useValue: createMock<KeycloakUserService>(),
                 },
+                {
+                    provide: ClassLogger,
+                    useValue: createMock<ClassLogger>(),
+                },
```
Comment (marode-cap) [inline src/modules/person/persistence/person.repository.integration-spec.ts:121]: Import LoggingTestModule

File: src/modules/personenkontext/api/dbiam-personenkontext-workflow.controller.ts [hunk 2]
```diff
@@ -243,23 +246,47 @@ export class DbiamPersonenkontextWorkflowController {
             params.personalnummer || undefined,
             params.befristung || undefined,
         );
+        if (savedPersonWithPersonenkontext instanceof Error) {
+            this.logger.error(
```
Comment (marode-cap) [inline src/modules/personenkontext/api/dbiam-personenkontext-workflow.controller.ts:250]: `instanceof Error` can have very weird behaviour in JS. Can we switch this to `!(savedPersonWithPersonenkontext instanceof PersonPersonenkontext)`?

File: src/modules/personenkontext/specification/personenkontext-specifications.integration-spec.ts [hunk 3]
```diff
@@ -99,6 +100,10 @@ describe('PersonenkontextSpecifications Integration', () => {
                     provide: RolleRepo,
                     useValue: createMock<RolleRepo>(),
                 },
+                {
+                    provide: ClassLogger,
+                    useValue: createMock<ClassLogger>(),
+                },
```
Comment (marode-cap) [inline src/modules/personenkontext/specification/personenkontext-specifications.integration-spec.ts:106]: Import LoggingTestModule

File: src/modules/privacy-idea-administration/privacy-idea-administration.controller.spec.ts [hunk 4]
```diff
@@ -52,6 +54,10 @@ describe('PrivacyIdeaAdministrationController', () => {
                     provide: PersonRepository,
                     useValue: createMock<PersonRepository>(),
                 },
+                {
+                    provide: ClassLogger,
+                    useValue: createMock<ClassLogger>(),
+                },
```
Comment (marode-cap) [inline src/modules/privacy-idea-administration/privacy-idea-administration.controller.spec.ts:60]: Import LoggingTestModule

**Review by marode-cap — CHANGES_REQUESTED** [review]

File: src/modules/personenkontext/api/dbiam-personenkontext-workflow.controller.ts (hunk 2 above)
Comment (DeFi3298) [reply src/modules/personenkontext/api/dbiam-personenkontext-workflow.controller.ts:250]: Wont work this way.
"The instanceof operator expects the right-hand operand to be a class or a constructor function, but here PersonPersonenkontext is likely a TypeScript interface or type alias, which is purely a compile-time construct and not available at runtime."

Comment (marode-cap) [reply src/modules/personenkontext/api/dbiam-personenkontext-workflow.controller.ts:250]: Ah, I thought it already was a class. In that case keep it like this, although I don't like how were returning multiple different things here (This should really be a `Result<>` instead, but it was already like this)

**Review by marode-cap — APPROVED** [review]

**Pull request merged**